### PR TITLE
Create iSCSI volumes with block size 4096 bytes

### PR DESCRIPTION
**Pull request by derekbit**
[Longhorn 4599](https://github.com/longhorn/longhorn/issues/4599)

TODO: update the go.mod go-iscsi-helpe, longhorn-engine and longhorn-instance-manager after review

Signed-off-by: Derek Su <derek.su@suse.com>

### CODE REVIEW

File: util/util.go [hunk 1]
```diff
@@ -895,3 +897,12 @@ func GetVolumeMeta(path string) (*VolumeMeta, error) {
 	}
 	return meta, nil
 }
+
+func GetVolumeBlockSize(blockSize int64) int64 {
+	// If blockSize is not defined, the existing volume uses 512 as its iSCSI LUN block size
+	if blockSize == 0 {
+		return 512
```
Comment (shuo-wu) [inline util/util.go:904]: NIT: Use const `LagacyBlockSize`

File: webhook/resources/engine/validator.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	}
+}
+
+func (e *engineValidator) Update(request *admission.Request, oldObj runtime.Object, newObj runtime.Object) error {
+	oldEngine := oldObj.(*longhorn.Engine)
+	newEngine := newObj.(*longhorn.Engine)
+
+	if oldEngine.Spec.BlockSize != newEngine.Spec.BlockSize {
+		return werror.NewInvalidError("Changing blockSize is not allowable", "")
+	}
```
Comment (shuo-wu) [inline webhook/resources/engine/validator.go:41]: Can we add an engine mutator to update `spec.blockSize` for existing engines?
Then this validator that prevents the **non-zero** `e.spec.blockSize` from being modified only.

File: util/util.go (hunk 1 above)
Comment (derekbit) [reply util/util.go]: Updated. Thank you.

File: webhook/resources/engine/validator.go (hunk 2 above)
Comment (derekbit) [reply webhook/resources/engine/validator.go]: Yes, updated. Thank you.

**Review by shuo-wu — APPROVED** [review]

**Pull request closed without merging**